src/fem/op_function.hh

```diff
@@ -0,0 +1,920 @@
+
+    void eval() override;
+};
+
+} // end of namespace Op::Side::El
+
+namespace Pt {
+
+} // end of namespace Op::Side::Pt
+
+} // end of namespace Side
+
+} // end of namespace Op
+
+
+
+/// Defines common functionality of reinit operations.
+struct common_reinit {
+	// empty base operation
+	static inline void op_base(FMT_UNUSED PatchOp<3> * result_op, FMT_UNUSED IntTableArena &el_table) {
+        // empty
+    }
+
+    template<unsigned int dim>
+    static inline void elop_jac(PatchOp<3> * result_op) {
+        // result matrix(spacedim, dim), input matrix(spacedim, dim+1)
+        auto jac_value = result_op->result_matrix();
+        auto coords_value = result_op->input_ops(0)->result_matrix();
+        for (unsigned int i=0; i<3; i++)
+            for (unsigned int j=0; j<dim; j++)
+                jac_value(i,j) = coords_value(i,j+1) - coords_value(i,0);
+    }
+
+    template<unsigned int dim>
+    static inline void elop_inv_jac(PatchOp<3> * result_op) {
+        // result matrix(spacedim, dim), input matrix(spacedim, dim+1)
+        auto inv_jac_value = result_op->result_matrix();
+        auto jac_value = result_op->input_ops(0)->result_matrix();
+        inv_jac_value = eigen_arena_tools::inverse<3, dim>(jac_value);
+    }
+
+    template<unsigned int dim>
+    static inline void elop_jac_det(PatchOp<3> * result_op) {
+        // result double, input matrix(spacedim, dim)
+        auto jac_det_value = result_op->result_matrix();
+        auto jac_value = result_op->input_ops(0)->result_matrix();
+        jac_det_value(0) = eigen_arena_tools::determinant<3, dim>(jac_value).abs();
+    }
+
+    static inline void ptop_JxW(PatchOp<3> * result_op) {
+        auto weights_value = result_op->input_ops(0)->result_matrix();
+        auto jac_det_value = result_op->input_ops(1)->result_matrix();
+        ArenaOVec<double> weights_ovec( weights_value(0,0) );
+        ArenaOVec<double> jac_det_ovec( jac_det_value(0,0) );
+        ArenaOVec<double> jxw_ovec = jac_det_ovec * weights_ovec;
+        auto jxw_value = result_op->result_matrix();
+        jxw_value(0,0) = jxw_ovec.get_vec();
+    }
+
+    /// Common reinit function of vector symmetric gradient on bulk and side points
+    static inline void ptop_vector_sym_grad(PatchOp<3> * result_op, FMT_UNUSED IntTableArena &el_table) {
+        for (uint i_dof=0; i_dof<result_op->n_dofs(); ++i_dof) {
+            Eigen::Map< Eigen::Matrix<ArenaVec<double>, Eigen::Dynamic, Eigen::Dynamic> > grad_vector_dof = result_op->input_ops(0)->result_sub_matrix(i_dof);
+            Eigen::Map< Eigen::Matrix<ArenaVec<double>, Eigen::Dynamic, Eigen::Dynamic> > sym_grad_dof = result_op->result_sub_matrix(i_dof);
+            sym_grad_dof = 0.5 * (grad_vector_dof.transpose() + grad_vector_dof);
+        }
+    }
+	/// Common reinit function of vector divergence on bulk and side points
+    static inline void ptop_vector_divergence(PatchOp<3> * result_op, FMT_UNUSED IntTableArena &el_table) {
```

Comment:
Pro společnou implementaci pro Bulk a Side, umístit do namespace Op:: a implementovat např. jako šablonu. To zajistí stejý kód ale různý identifikátor třídy.